### ISSUE TITLE
handleRedirectPromise returning null with new sessions on Safari.

### ISSUE THREAD

**Issue by MagnusHJensen** [AzureAD/microsoft-authentication-library-for-js#4444]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.21.0

### Wrapper Library

MSAL React (@azure/msal-react)

### Wrapper Library Version

1.1.0

### Description
_Not reproducible with Safari 15.2._

Accessing the application url, works the very first time when not signed in. 

Any subsequent hits to the same application in a new tab/session, will break on safari saying the `handleRedirectPromise no interaction is in progress, returning null` (The autherization code is retrieved and is in the url)

Refreshing that same session after failing, will load normally.

I have tried disabling the "Prevent Cross-Site tracking", we have also tried using localStorage instead of sessionStorage. While that works over the same browser, any new browser/user needs to go through this issue first, and then it works, which is not optimal.

### Msal Logs

[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Event callback registered with id: cc0a66cd-1bb3-4719-8293-d780888d3cfe (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Verbose - useMsalAuthentication - Registered event callback with id: cc0a66cd-1bb3-4719-8293-d780888d3cfe (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Event callback registered with id: a385bf30-088b-45af-8f96-bb71823d2076 (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Verbose - MsalProvider - Registered event callback with id: a385bf30-088b-45af-8f96-bb71823d2076 (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Event callback registered with id: f462be40-b299-4e33-a1a0-4474337e96ee (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Verbose - MsalProvider - Registered event callback with id: f462be40-b299-4e33-a1a0-4474337e96ee (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - handleRedirectPromise called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:handleRedirectStart (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback cc0a66cd-1bb3-4719-8293-d780888d3cfe: msal:handleRedirectStart (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback a385bf30-088b-45af-8f96-bb71823d2076: msal:handleRedirectStart (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback f462be40-b299-4e33-a1a0-4474337e96ee: msal:handleRedirectStart (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Info - MsalProvider - msal:handleRedirectStart results in setting inProgress from startup to handleRedirect (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - handleRedirectPromise has been called for the first time, storing the promise (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [be33c73e-4b0f-474d-a92e-7e624f102f8a] : msal.js.browser@2.21.0 : Verbose - initializeServerTelemetryManager called (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : [be33c73e-4b0f-474d-a92e-7e624f102f8a] : msal.js.browser@2.21.0 : Info - handleRedirectPromise called but there is no interaction in progress, returning null. (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857, x2)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:handleRedirectEnd (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback cc0a66cd-1bb3-4719-8293-d780888d3cfe: msal:handleRedirectEnd (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback a385bf30-088b-45af-8f96-bb71823d2076: msal:handleRedirectEnd (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Info - MsalProvider - no account changes (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback f462be40-b299-4e33-a1a0-4474337e96ee: msal:handleRedirectEnd (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Info - MsalProvider - msal:handleRedirectEnd results in setting inProgress from handleRedirect to none (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Info - useMsalAuthentication - No user is authenticated, attempting to login (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Verbose - useMsalAuthentication - Calling loginRedirect (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - loginRedirect called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - preflightBrowserEnvironmentCheck started (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - acquireTokenRedirect called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:loginStart (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback cc0a66cd-1bb3-4719-8293-d780888d3cfe: msal:loginStart (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback a385bf30-088b-45af-8f96-bb71823d2076: msal:loginStart (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - Emitting event to callback f462be40-b299-4e33-a1a0-4474337e96ee: msal:loginStart (main.chunk.js, line 7857)
[Info] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-react@1.1.0 : Info - MsalProvider - msal:loginStart results in setting inProgress from none to login (main.chunk.js, line 7853)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - preflightInteractiveRequest called, validating app environment (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - initializeAuthorizationRequest called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - getRedirectUri called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - Initializing BaseAuthRequest (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - Authentication Scheme wasn't explicitly set in request, defaulting to "Bearer" request (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857, x2)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - initializeServerTelemetryManager called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - initializeAuthorizationRequest called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - getClientConfiguration called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - getDiscoveredAuthority called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - Creating discovered authority with request authority (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - Auth code client created (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - RedirectHandler.initiateAuthRequest called (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - RedirectHandler.initiateAuthRequest: redirectStartPage set, caching start page (main.chunk.js, line 7857)
[Debug] [Tue, 25 Jan 2022 11:25:58 GMT] : [de92658d-c666-4fa2-a392-fe2ec0bcbee5] : msal.js.browser@2.21.0 : Verbose - RedirectHandler.initiateAuthRequest: Navigating window to navigate url (main.chunk.js, line 7857)

### MSAL Configuration

```javascript
export const loginType: InteractionType = InteractionType.Redirect;

export const msalConfig = {
  auth: {
    clientId: env.msClientId,
    authority: env.msAuthorityId,
    redirectUri: '/',
  },
  cache: {
    cacheLocation: 'sessionStorage', // This configures where your cache will be stored
    storeAuthStateInCookie: false, // Set this to "true" if you are having issues on IE11 or Edge
  }
}
```

### Relevant Code Snippets

Use the standard msal-react and msal-browser features, to get it up and a default msalConfig.

### Reproduction Steps

Make a basic react app, add the `MsalProvider` componentn with the `PublicClientApplication(msalConfig)` as the instance.
Wrap your gated content with the `MsalAuthenticationTemplate` with InteractionType.Redirect and no additional scopes in the loginRequest.
Navigate to the application and authorise.
Open a new session and navigate to the application again on safari, and the issue comes up.

### Expected Behavior

Navigating to the same application in a new session on safari, should load the app and not fail after retrieving the auth code.

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Safari version: (15.1 (17612.2.9.1.20))

### Source

External (Customer)

**Comment by HassanMahmud** [comment]
[Disclaimer - i'm on the same team as Magnus]

A user has reported a similar experience of the page getting stuck after the redirect on Edge. 
I will confirm asap if it is indeed this exact issue or something else.

**Comment by jasonnutter** [comment]
@MagnusHJensen Is this similar to #4458?

**Comment by MagnusHJensen** [comment]
It is indeed.

Same "fix" works.
Also looks like it was fixed with Safari 15.2, so not sure what changed between .1 and .2.
